Source/main.cpp

```diff
@@ -1,36 +1,217 @@
-
-#include "../Header/graph.h"
+#include "../Header/graph_color.h"
+#include "../Header/dsatur.h"
+#include "../Header/mcs.h"
+#include "../Header/lmxrlf.h"
+#include "../Header/hybrid_dsatur.h"
+#include "../Header/hybrid.h"
+#include <fstream>
 #include <iostream>
+#include <sstream>
 
 using std::cerr;
 using std::cout;
 using std::endl;
-using GraphColoring::Graph;
+using std::ifstream;
+using std::ofstream;
+using std::ostringstream;
+using GraphColoring::dsatur;
+using GraphColoring::mcs;
+using GraphColoring::lmxrlf;
+using GraphColoring::hybrid_dsatur;
+using GraphColoring::hybrid;
+using GraphColoring::GraphColor;
 using GraphColoring::Algorithm;
 
+//functions used to graphs test cases to map
+vector<string> split(string to_split);
+vector< vector<string> > get_input(char* input_file);
+void parse_edge_list(char* input_file);
+void parse_edge_matrix(char* input_file);
+
+map<string,vector<string> > input_graph;
+
 int main(int argc, char** argv)
 {
     if(argc < 2) {
         cerr << "Usage: " << argv[0] << " file_input" << endl;
         return -1;
     }
-    
-    Graph graph(GraphColoring::kMCS);
-    
+
     if(argc >= 3 && string(argv[2]) == "-m") {
-        graph.parse_edge_matrix(argv[1]);
-        if(graph.size() == 0) { return -2; }
+        parse_edge_matrix(argv[1]);
+        if(input_graph.size() == 0) { return -2; }
     } else if(argc >= 3 && string(argv[2]) == "-l") {
-        graph.parse_edge_list(argv[1]);
-        if(graph.size() == 0) { return -2; }
+        parse_edge_list(argv[1]);
+        if(input_graph.size() == 0) { return -2; }
     } else {
         cout << "No Graph Input Type Selected" << endl;
         return -1;  }
-    
-    graph.color();
-    graph.print_chromatic();
-    graph.verify();
-    graph.write_graph();
+
+    GraphColor *graph = new dsatur(input_graph);
+
+    graph->color();
+    graph->print_chromatic();
+    graph->verify();
+    graph->write_graph();
 
     return 0;
 }
+
```

Comment:
When I first started wrapping all the algorithms, you wanted me to take the parsing and graph creation out of the graph_coloring class. Did you want just the parsing functions back in the class?